docs/ADR/2019-11-27-firestore-saving-timed-transcripts.md

```diff
@@ -0,0 +1,227 @@
+* Deciders: Eimi, Pietro <!-- optional -->
+* Date: 2019-11-27  <!-- optional -->
+
+Technical Story: [description | ticket/issue URL] <!-- optional -->
+
+## Context and Problem Statement
+
+### the Conext
+The context is the development of applications to work with transcriptions of audio or video interviews, these could range from 20 minutes to 1 hour or over in length, depending on the production requirements.
+
+
+### The Problem 
+Saving word level timed text transcriptions into firebase.
+
+Data structure (refered to as DPE Json)
+
+```json
+{
+  "words": [
+    {
+      "id": 0,
+      "start": 1.4,
+      "end": 3.9,
+      "text": "Hello"
+    },
+    ...
+  ],
+  "paragraphs": [
+    {
+      "id": 0,
+      "start": 1.4,
+      "end": 4.2,
+      "speaker": "TBC 0"
+    },
+  ...
+  ]
+}
+```
+
+Paragraphs and speakers are interpolated on the client.
+
+In this domain we generally use components such as [@bbc/react-transcript-editor](https://github.com/bbc/react-transcript-editor) as a way to manipulate timed text.
+
+Generally this data structure is quite efficient, compared to GCP STT one hour worth of transcript, in GCP STT is easily 2mb while in this DPE format it is 1.3mb.
+
+We have seen that 20 min worth of timed text, can be saved in firestore, as in this DPE json format it's roughly 315 KB. - roughly 3581 words.
+
+_For an example see [Soleio interiview as example in DPE demo](https://bbc.github.io/digital-paper-edit-client/#/projects/10046281c4ad4938b7d0ae6fa9899bec/transcripts/1000cjw29xii80000ird74yb19swa/correct), (click export btn arrow top right, and choose last option `Digital Paper Edit - Json`)_
+
+## Decision Drivers <!-- optional -->
+
+1. Easy to reason around
+2. Keeping cost down
+3. Firestore document size limit of 1mb 
+4. In Firestore reads of documents are charged individually 
+5. Ideally a solution that can scale for storing over 1 hour worth of transcription without breaking a sweat
+6. if needed paragraphs and words attribute could be saved in separate collections/documents.
+
+
+## Considered Options
+
+1. Break down data in firestore - "paginated" every 20 min
+2. Break down data in firestore - "paginated" every paragraph
+3. Break down data in firestore - every word is a document
+4. Saving json in Firebase cloud storage?
+5. Compression - JSONB 
+6. Compression - JSONC
+7. Compression - binary with UBJSON
+
+## Decision Outcome
+
+_TBC_
+<!-- Chosen option: "[option 1]", because [justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)]. -->
+
+<!-- ### Positive Consequences 
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative consequences
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* … -->
+
+## Pros and Cons of the Options <!-- optional -->
+
+### 1. Break down data in firestore - "paginated" every 20 min
```

Comment:
> We have seen that 20 min worth of timed text transcription in JSON (see below, referred to as DPE JSON) format is roughly 315 KB - roughly 3581 words.

I would argue that 20 minutes is conservative - would probably push it up to 30 minutes. I think if the limit is 1000KB, we should at least aim to be around the 50% mark (500KB).